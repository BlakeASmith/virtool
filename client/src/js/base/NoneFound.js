import React from "react";
import PropTypes from "prop-types";
import { ListGroup, ListGroupItem } from "react-bootstrap";
import { Icon } from "./Icon";

/**
 * A ListGroupItem component with a 'none found'-type message. Used in ListGroups when no data is available to populate
 * the list. For example, when no sample have been created.
 *
 * @param noun {string} the name of the items of which none were found (eg. samples)
 * @param noListGroup {boolean} don't include a ListGroup in the returned element
 */
export const NoneFound = ({ noun, noListGroup, style }) => {
  const item = (
    <ListGroupItem className="text-center">
      <Icon name="info-circle" /> No {noun} found
    </ListGroupItem>
  );

  if (noListGroup) {
    return item;
  }

  return <ListGroup style={style}>{item}</ListGroup>;
};

NoneFound.propTypes = {
  noun: PropTypes.string.isRequired,
  noListGroup: PropTypes.bool,
  style: PropTypes.object
};
