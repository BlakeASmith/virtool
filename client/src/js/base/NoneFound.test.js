import React from "react";
import { NoneFound } from "./NoneFound";
import { ListGroup, ListGroupItem } from "react-bootstrap";

describe("<NoneFound />", () => {
    let wrapper;

    describe("when supplied [noListGroup=false] prop", () => {
        const noun = "test";
        const noListGroup = false;

        wrapper = shallow(<NoneFound noun={noun} noListGroup={noListGroup} />);

        it("renders correctly", () => {
            expect(wrapper).toMatchSnapshot();
        });

        it("renders a ListGroup", () => {
            expect(wrapper.find(ListGroup).exists()).toBe(true);
        });
    });

    describe("when supplied [noListGroup=true] prop", () => {
        const noun = "test";
        const noListGroup = true;

        wrapper = shallow(<NoneFound noun={noun} noListGroup={noListGroup} />);

        it("renders correctly", () => {
            expect(wrapper).toMatchSnapshot();
        });

        it("renders a ListGroupItem without ListGroup container", () => {
            expect(wrapper.find(ListGroupItem).exists()).toBe(true);
        });
    });

});
