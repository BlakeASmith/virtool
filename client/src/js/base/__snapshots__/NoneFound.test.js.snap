// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoneFound /> when supplied [noListGroup=false] prop renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoneFound
    noListGroup={true}
    noun="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "list-group-item",
      "children": Array [
        <Icon
          fixedWidth={false}
          name="info"
          pending={false}
          pullRight={false}
        />,
        " No ",
        "test",
        " found",
      ],
      "className": "text-center",
      "listItem": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fixedWidth": false,
          "name": "info",
          "pending": false,
          "pullRight": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " No ",
      "test",
      " found",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "list-group-item",
        "children": Array [
          <Icon
            fixedWidth={false}
            name="info"
            pending={false}
            pullRight={false}
          />,
          " No ",
          "test",
          " found",
        ],
        "className": "text-center",
        "listItem": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fixedWidth": false,
            "name": "info",
            "pending": false,
            "pullRight": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " No ",
        "test",
        " found",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<NoneFound /> when supplied [noListGroup=true] prop renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoneFound
    noListGroup={true}
    noun="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "list-group-item",
      "children": Array [
        <Icon
          fixedWidth={false}
          name="info"
          pending={false}
          pullRight={false}
        />,
        " No ",
        "test",
        " found",
      ],
      "className": "text-center",
      "listItem": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fixedWidth": false,
          "name": "info",
          "pending": false,
          "pullRight": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " No ",
      "test",
      " found",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "list-group-item",
        "children": Array [
          <Icon
            fixedWidth={false}
            name="info"
            pending={false}
            pullRight={false}
          />,
          " No ",
          "test",
          " found",
        ],
        "className": "text-center",
        "listItem": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fixedWidth": false,
            "name": "info",
            "pending": false,
            "pullRight": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " No ",
        "test",
        " found",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
